### PR TITLE
refactor: enhance UVE script injection logic 

### LINKED ISSUE #35127: UVE editor fails to load after v26.03.20-01 — dot-uve.js script injection regression
## Problem Statement

After upgrading to version **v26.03.20-01**, the Universal Visual Editor (UVE) stops loading on all pages for all users. The edit/add content buttons do not appear, making the authoring environment completely unusable.

**Root cause:** In v26.03.20-01, the `dot-uve.js` script injection mechanism was moved from the Angular frontend to the Java backend via two paired commits:

- **PR #34927** (`89e5da9e7b`): Backend now injects `<script src="/ext/uve/dot-uve.js">` into `entity.page.rendered` for non-LIVE modes via `HTMLPageAssetRenderedBuilder.injectUVEScript()`
- **PR #34995** (`6b87762eda`): Frontend removed its own script injection (`addEditorPageScript` method in `EditEmaEditorComponent`)

If the backend fails to inject the script — due to cache, proxy/CDN stripping, deploy skew (only one of the two PRs deployed), or an exception in `buildUVEStyleEditorScripts()` — the `SET_CONTENTLET` event is never fired and the floating edit/add UI never appears.

**Impact:** Critical — affects all users, all pages, both authoring and production environments. Customer-reported production outage (Freshdesk #36029).

**Browser & OS:** All browsers/OS — issue is server-side.

## Steps to Reproduce

1. Upgrade a dotCMS instance from v26.03.17-01 to v26.03.20-01
2. Log in to the admin UI and navigate to any page in the UVE editor (Edit mode)
3. Hover over any contentlet on the page
4. **Expected:** `dot-uve-contentlet-tools` (edit/add buttons) appear on hover
5. **Actual:** No edit/add controls appear; the page renders but is not editable

**Verification:**
- Open browser DevTools → Elements tab → search for `dot-uve.js` in the iframe document
- If the `<script src="/ext/uve/dot-uve.js"></script>` tag is missing from the rendered HTML, the backend injection failed
- Also check the Network tab for 403 errors on `/ext/uve/dot-uve.js` (CDN/proxy blocking)

**Key files:**
- Backend injection: `dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRenderedBuilder.java` — `injectUVEScript()` method
- Frontend (script removed): `core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts` — `#insertPageContent()`
- API flow: `dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java` — `getPageRender()`

## Acceptance Criteria

**Acceptance criteria** — To be refined. (Quick draft; details to be added later.)

- [ ] `dot-uve.js` script tag is present in `entity.page.rendered` for EDIT_MODE and PREVIEW_MODE
- [ ] `dot-uve.js` script tag is NOT present in `entity.page.rendered` for LIVE mode
- [ ] `buildUVEStyleEditorScripts()` failures fall back gracefully to the plain `SDK_EDITOR_SCRIPT_SOURCE` tag (defensive Try.of wrapper)
- [ ] Diagnostic logging added to `injectUVEScript` to trace injection/skip decisions
- [ ] Integration tests cover edge cases: empty containers, null ContentType metadata, exception in schema stream
- [ ] UVE edit/add buttons appear correctly after upgrade from v26.03.17-01 to the fixed version

## dotCMS Version

- **Affected:** v26.03.20-01
- **Last known good:** v26.03.17-01
- **Relevant commits between versions:**
  - `89e5da9e7b` — PR #34927 (backend: inject UVE script into page.rendered)
  - `6b87762eda` — PR #34995 (frontend: remove UVE script injection)
  - `9c0834db27` — PR #35046 (post-release: added style editor schema support to injection, changed `injectUVEScript` signature)

## Severity

Critical - System unusable

## Links

- [Freshdesk ticket #36029](https://dotcms.freshdesk.com/helpdesk/tickets/36029)

### PR DESCRIPTION
**Pull request by nicobytes**
This pull request improves the logging and error handling in the UVE script injection process within the HTML page rendering flow. The changes make it easier to trace when and how the UVE script is injected, and ensure that failures in script building are handled gracefully.

**Enhanced logging and error handling for UVE script injection:**

* Added debug logs to indicate when the UVE script is injected or skipped based on the page mode in the `build` method of `HTMLPageAssetRenderedBuilder` (`dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRenderedBuilder.java`).
* Improved the `injectUVEScript` method to:
  - Log when the rendered HTML is empty or null and script injection is skipped.
  - Use a `Try` block to handle exceptions during script building, logging errors and falling back to a default script if needed.
  - Add debug logs to indicate where the script is injected (before `</body>` or at the end if no `</body>` tag is found).

#35127

### CODE REVIEW

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRenderedBuilder.java [hunk 1]
```diff
@@ -388,13 +398,22 @@ private void transformLegacyContainerUUIDs(TemplateLayout layout) {
      */
     private String injectUVEScript(final String html, final Collection<? extends ContainerRaw> containers) {
         if (!UtilMethods.isSet(html)) {
+            Logger.debug(this, "Skipping UVE script injection: rendered HTML is empty or null");
             return html;
         }
-        final String scripts = buildUVEStyleEditorScripts(containers).orElse(SDK_EDITOR_SCRIPT_SOURCE);
+        final String scripts = Try.of(() -> buildUVEStyleEditorScripts(containers)
```
Comment (dario-daza) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRenderedBuilder.java]: I reckon that a Try.of here for a log is not worthy I would do the login like this:
```suggestion
        final Optional<String> styleEditorScript = buildUVEStyleEditorScripts(containers);
        final String scripts = styleEditorScript.orElse(SDK_EDITOR_SCRIPT_SOURCE);
        Logger.debug(this, () -> styleEditorScript.isPresent()
                ? "Injecting UVE script with style editor schemas"
                : "Injecting plain UVE script (no style editor schemas found)");
```

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRenderedBuilder.java [hunk 2]
```diff
@@ -388,13 +398,22 @@ private void transformLegacyContainerUUIDs(TemplateLayout layout) {
...
+                        .orElse(SDK_EDITOR_SCRIPT_SOURCE))
+                .onFailure(e -> Logger.error(this,
+                        "Failed to build UVE style editor scripts, falling back to plain script tag", e))
+                .getOrElse(SDK_EDITOR_SCRIPT_SOURCE);
         final int closingBodyIndex = html.toLowerCase().lastIndexOf("</body>");
         if (closingBodyIndex != -1) {
+            Logger.debug(this, () -> String.format(
```
Comment (dario-daza) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRenderedBuilder.java]: I would omit this log, I think is redundant, the log below already describes the position of the log, our main concern is more about if the UVE script is injected

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRenderedBuilder.java [hunk 3]
```diff
@@ -388,13 +398,22 @@ private void transformLegacyContainerUUIDs(TemplateLayout layout) {
...
+                    "UVE script injected before </body> (HTML length: %d)", html.length()));
             return html.substring(0, closingBodyIndex) + scripts + html.substring(closingBodyIndex);
         }
+        Logger.debug(this, () -> String.format(
```
Comment (dario-daza) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRenderedBuilder.java]: Printing the html.length() is low value, this log covers the one that you put above:
`UVE script injected before </body>`

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRenderedBuilder.java [hunk 4]
```diff
@@ -215,9 +215,19 @@ public PageView build(final boolean rendered, final PageMode mode) throws DotDat
             final Collection<? extends ContainerRaw> containers = new ContainerRenderedBuilder(
                     pageRenderUtil.getContainersRaw(), velocityContext, mode)
                     .build();
-            final String pageHTML = mode != PageMode.LIVE
-                    ? injectUVEScript(this.getPageHTML(mode), containers)
-                    : this.getPageHTML(mode);
+            final String rawHTML = this.getPageHTML(mode);
+            final String pageHTML;
+            if (mode != PageMode.LIVE) {
+                Logger.debug(this, () -> String.format(
```
Comment (dario-daza) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRenderedBuilder.java:222]: Cool 👍

**Review by dario-daza — APPROVED** [review]

**Review by freddyDOTCMS — APPROVED** [review]

**Review by dario-daza — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]
